### PR TITLE
chore: enforce drizzle usage

### PR DESCRIPTION
**Pull request by elbalexandre**
## Description

- enforce `drizzle` usage in `CONTRIBUTION.md`
- configure `drizzle` in the template
- configure `vitest` to works along the db in the template
- update env. variables template files
- add documentation for tests in `CONTRIBUTION.md`
- bump template vitest version to `beta-5` (required for running test in a single thread properly).
- add `docker-compose.yml` to setup a local database and proxy
- add script for database migration
- add script to run the tests with a database (working on pipeline)

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: template/.env.local.example
[the remark this answers is not in the document]
Comment (elbalexandre) [reply template/.env.local.example:1]: The issue is that the gitignore have to be in the template as well.

File: CONTRIBUTING.MD [hunk 1]
```diff
@@ -9,6 +9,22 @@ The following is a set of guidelines for contributing to **elba-security** and i
...
+
+As Vercel offers storage hosting, we chose to select one of the supported databases. While a relational database was not a requirement, some integrations could benefit from this feature.
+
+#### `drizzle-orm`
+
+We are enforcing the use of [`drizzle-orm`](https://orm.drizzle.team/docs/overview), which provides type safety and migration generation. Unlike the `@vercel/postgres` utility `sql` used alone, it allows for local database usage.
+
+#### `vitest`
+
+We are requiring [`vitest`](https://vitest.dev/) as the test runner for several reasons. It is well-maintained, and its configuration is less tricky than that for `jest`. Additionally, it is faster.
```
[the remark this answers is not in the document]
Comment (athielba) [reply CONTRIBUTING.MD:24]: Ignorable:  Our beloved friend ChatGpt Version 😆 
We have opted to utilize [`vitest`](https://vitest.dev/) as our preferred test runner for a number of compelling reasons. This choice is primarily driven by its commendable track record of being well-maintained, as well as its straightforward configuration, which stands in contrast to the somewhat intricate setup required by `jest`. Furthermore, `vitest` demonstrates superior performance in terms of speed.

**Comment by athielba** [comment]
@elbalexandre  Nice work  👏🏼

**Review by proquig — APPROVED** [review]
Looks good just a small comment

File: template/.env.test [hunk 2]
```diff
@@ -1 +1,11 @@
+ELBA_SOURCE_ID="0442f541-45d2-487a-9e4b-de03ce4c559e"
+ELBA_API_KEY="abc123dfg456"
 CRON_SECRET="1234"
+POSTGRES_URL="postgres://postgre:postgre@localhost:5556/test"
+POSTGRES_HOST="0.0.0.0"
+POSTGRES_PORT="5556"
+POSTGRES_USERNAME="postgre"
```
[the remark this answers is not in the document]
Comment (elbalexandre) [reply template/.env.test]: thx, it's fixed

**Pull request merged**